subsys/usb/usb_c/usbc_stack.c

```diff
@@ -128,7 +133,12 @@ int usbc_request(const struct device *dev, const enum usbc_policy_request_t req)
 
 	/* Add public request to fifo */
 	data->request.val = req;
-	k_fifo_put(&data->request_fifo, &data->request);
+	if (k_fifo_is_empty(&data->request_fifo)) {
```

Comment:
You should return an appropriate error code, informing the caller that the request was not successful. Also, how do you know that the same request is being added?

Reply:
This case was a misuse of the usbc_request function by my changes. I have added function dedicated to preventing one sleep of USB-C thread causing the one more iteration of the state machines.